ttnn/cpp/ttnn/operations/eltwise/binary_ng/device/binary_ng_program_factory.cpp

```diff
@@ -475,10 +475,10 @@ BinaryNgDeviceOperation::ProgramFactory::cached_program_t BinaryNgDeviceOperatio
                 compute_kernel_defines["QUANT_ZERO_POINT_RT_ARGS_IDX"] = "3";
                 unary::utils::update_macro_defines(unary::UnaryOpType::ZERO_POINT, compute_kernel_defines);
             } else {
-                add_activation_defines(compute_kernel_defines, post_activations, "POST");
+                add_activation_defines(compute_kernel_defines, post_activations, "POST", a_dtype);
             }
         } else {
-            add_activation_defines(compute_kernel_defines, post_activations, "POST");
+            add_activation_defines(compute_kernel_defines, post_activations, "POST", a_dtype);
```

Comment:
Pretty sure these should be `c.get_dtype()`, right? Let's add a test case to cover a datatype-dependent unary post activation.